### PR TITLE
ENH: Use tz-aware dtype for timestamp columns

### PR DESCRIPTION
**Pull request by tswast**
I couldn't figure out how *not* to get a tz-aware dtype in 0.24.x
versions, and I wanted a tz-aware dtype anyway for TIMESTAMP, so this
makes it official.

Towards #261.

### CODE REVIEW

**Comment by tswast** [comment]
Looks like we'll have to bump the minimum version to `0.20.0`, as `pandas.api.types.DatetimeTZDtype` was first exposed publicly in https://github.com/pandas-dev/pandas/pull/16099

File: docs/source/reading.rst [hunk 1]
```diff
@@ -37,20 +48,39 @@ your job. For more information about query configuration parameters see `here
...
+
+================== =========================
+BigQuery Data Type dtype
+================== =========================
+FLOAT              float
+TIMESTAMP          DatetimeTZDtype(unit='ns', tz='UTC')
+DATETIME           datetime64[ns]
+TIME               datetime64[ns]
+DATE               datetime64[ns]
+================== =========================
```
Comment (max-sixty) [inline docs/source/reading.rst]: 👍

**Comment by tswast** [comment]
Oof. Looks like I still get the occasional `TypeError: data type not understood` for all versions before 0.24.0.

@max-sixty Thoughts on bumping the minimum version all the way to 0.24? It's a big jump from what we've had before. Otherwise, I'll see if I can make our tests robust against having naive datetime in 0.23 and before.

**Comment by tswast** [comment]
I'm going to branch this PR with just the docs changes. I'd rather not bump all the way to 0.24.0 just yet.

**Comment by tswast** [comment]
Closing in favor of https://github.com/pydata/pandas-gbq/pull/269. As was pointed out to me in https://github.com/pandas-dev/pandas/issues/25843#issuecomment-476426843 it's more "idiomatic" to call localize rather than be specific about timezones in the dtype. I'd prefer to avoid additional casts / localization if at all possible, so just documenting the current behavior for now.

**Pull request closed without merging**